hubblestack_nova/stat.py

```diff
@@ -0,0 +1,141 @@
+from __future__ import absolute_import
+import logging
+
+import fnmatch
+import yaml
+import os
+import copy
+import salt.utils
+
+from distutils.version import LooseVersion
+
+log = logging.getLogger(__name__)
+
+__tags__ = None
+__data__ = None
+
+
+def __virtual__():
+    if salt.utils.is_windows():
+        return False, 'This audit module only runs on linux'
+    global __tags__
+    global __data__
+    yamldir = os.path.dirname(__file__)
+    if not yamldir:
+        yamldir = '.'
+    __data__ = _get_yaml(yamldir)
+    __tags__ = _get_tags(__data__)
+    return True
+
+
+def audit(tags, verbose=False):
+    '''
+    Run the grep audits contained in the YAML files processed by __virtual__
+    '''
+    ret = {'Success': [], 'Failure': []}
+
+    for tag in __tags__:
+        if fnmatch.fnmatch(tag, tags):
+            for tag_data in __tags__[tag]:
+                name = tag_data['name']
+                expected = {}
+                for e in ['mode', 'user', 'group']:
+                    if e in tag_data:
+                        expected[e] = str(tag_data[e])
+
+                #getting the stats using salt
+                salt_ret = __salt__['file.stats'](name)
+                if not salt_ret:
+                    if 'cannot stat' in expected.values():
+                        ret['Success'].append(tag_data)
+                    else:
+                        ret['Failure'].append(tag_data)
+                    continue
+
+                passed = True
+                for e in expected.keys():
+                    r = salt_ret[e]
+                    if e == 'mode' and r != '0':
+                        r = r[1:]
+                    if expected[e] != r:
```

Comment:
Since I want to support uid/gid in addition to user/group, I think we should force this section to strings.

```
if str(expected[e]) != str(r):
```

Otherwise we'll end up comparing `0` and `'0'` and run into problems.